### ISSUE TITLE
Change controller capabilities from compiled  to /directory/config file read on startup. 

### ISSUE THREAD

**Issue by stevelawrence66** [smeighan/xLights#4108]
I'd like to suggest an enhancement to the way controller capabilities are read in. 
Currently compiled from /controllers/[name].controller 

This makes it hard for folks "making their own" to provide their own capabilities. 
It also means that an update to xLights is required if/when new controllers come out ( eg: Experience Lights G16, G16 Pro, Falcon V5). 

Since the capabilities should *really* be owned by the board creator, not the xLights devs, I'd like to have a /controllers directory on install, and a 'default' set of controllers dumped in there ( your usual suspects). Read that directory for .controller files on startup to get your 'board capabilities' and that way if I make a new board, I can simply put my own thing.controller file in there to define its capabilities.

Likewise, vendors who come out with new boards mid-season etc.. can provide a .controller file for users to dump into that directory and immediately gain access to their new controller, instead of waiting on an xLights dev to push a new version, which may or may not have features which break, differ, or are incompatible with their 'pinned version'. 

This way vendors/users  can be responsible for their boards and capabilities, instead of 'behind the scenes' hard-coded values.

**Comment by mjunek** [comment]
You can do this already. The xController files are not part of the compilation process, they are read on startup.
I have a file that I copy from install to install for a one-off board that I have made.

**Comment by stevelawrence66** [comment]
Thanks Mike, I failed to spot this before submitting. :)
